### PR TITLE
Tsan v1.1

### PR DESCRIPTION
**Pull request by rf972**
The only changes we need to review here are the first one's configure changes to error out if tsan not available, and the last commit's testing.rst changes.

### CODE REVIEW

**Review by PeterPuhov-Linaro — APPROVED** [review]

File: docs/devel/testing.rst [hunk 1]
```diff
@@ -397,6 +397,118 @@ list is in the ``make docker`` help text. The frequently used ones are:
...
+First, to configure the build for TSan:
+
+.. code::
+
+  ../configure --enable-tsan --cc=clang-10 --cxx=clang++-10 \
+               --disable-werror --extra-cflags="-O0"
+
+There is also a one time patch needed in clang-9 or clang-10 at this time:
+
+.. code::
```
Comment (PeterPuhov-Linaro) [inline docs/devel/testing.rst:446]: I would suggest to mention clang patch first.
People usually run examples in order :)

File: docs/devel/testing.rst [hunk 2]
```diff
@@ -397,6 +397,118 @@ list is in the ``make docker`` help text. The frequently used ones are:
...
+
+"--extra-cflags=-fsanitize-blacklist=<src path>/tests/tsan/blacklist.tsan"
+
+More information on the file format can be found here under "Blacklist Format":
+
+https://github.com/google/sanitizers/wiki/ThreadSanitizerFlags
+
+TSan Annotations
+----------------
+include/qemu/tsan.h defines annotations.  See this file for more descriptions
```
Comment (PeterPuhov-Linaro) [inline docs/devel/testing.rst:497]: Are we planning to include this file in our patch ?

**Review by PeterPuhov-Linaro — APPROVED** [review]

File: docs/devel/testing.rst (hunk 1 above)
Comment (rf972) [reply docs/devel/testing.rst:446]: Sure, makes sense.  Will change it.

File: docs/devel/testing.rst (hunk 2 above)
Comment (rf972) [reply docs/devel/testing.rst:497]: This file, tsan.h is in the patch series.  It is the 3rd from the last patch.

**Pull request closed without merging**